test/tests/selftest/pbs_syslog_tests.py

```diff
@@ -0,0 +1,146 @@
+        """
+        self.setup_syslog(local_log=0, syslog_facility=9, syslog_severity=7)
+        a = {'Resource_List.ncpus': 1}
+        J1 = Job(TEST_USER, attrs=a)
+        jid1 = self.server.submit(J1)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        msg_str = "Job;" + jid1 + ";Job Run at request of Scheduler"
+        self.server.log_match(msg_str, n=500)
+
+    def test_basic_syslog_severity(self):
+        """
+        Test that that PBS logs messages via syslog
+        according to the set priority
+        :return:
+        """
+        self.t = int(time.time())
+        self.setup_syslog(local_log=0, syslog_facility=9, syslog_severity=6)
+        a = {'Resource_List.ncpus': 1}
+        J1 = Job(TEST_USER, attrs=a)
+        jid1 = self.server.submit(J1)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        msg_str = "Job;" + jid1 + ";Considering job to run"
+        with self.assertRaises(PtlLogMatchError):
+            self.scheduler.log_match(
+                msg_str, n=500, max_attempts=10, starttime=self.t)
+        self.server.deljob(jid1, wait=True)
+        self.setup_syslog(local_log=0, syslog_facility=9, syslog_severity=7)
+        J2 = Job(TEST_USER, attrs=a)
+        jid2 = self.server.submit(J2)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid2)
+        msg_str = "Job;" + jid2 + ";Considering job to run"
+        self.scheduler.log_match(msg_str, n=500, max_attempts=10)
+
+    def test_muti_host_syslog_match(self):
+        """
+        Test that syslog matching works on a multihost system
+        :return:
+        """
+        self.setup_syslog(local_log=0, syslog_facility=9, syslog_severity=7)
+        if len(self.moms) != 2:
+            self.skip_test(reason="need 2 mom hosts: -p moms=<m1>:<m2>")
+
+        self.momA = self.moms.values()[0]
+        self.momB = self.moms.values()[1]
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+
+        attr = {'Resource_List.select': '2:ncpus=1'}
+        j = Job(TEST_USER, attrs=attr)
+        jid1 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        msg_str = "Job; " + jid1 + ";Started"
+        self.momA.log_match(msg_str, starttime=self.t, n=200)
+        self.momB.log_match(msg_str, starttime=self.t, n=200)
+
+    def test_local_and_syslog_match(self):
+        """
+        Test that local logging and sylog logging can work simultaneously
+        """
+        self.setup_syslog(local_log=1, syslog_facility=9, syslog_severity=7)
+        self.t = int(time.time())
+        a = {'Resource_List.ncpus': 1}
+        J1 = Job(TEST_USER, attrs=a)
+        jid1 = self.server.submit(J1)
+        # self.server.log_match(jid2,
+        #                      max_attempts=1)
+        msg_str = "Job;" + jid1 + ";Job Queued at request of "
+        self.server.log_match(msg_str,
+                              max_attempts=1, starttime=self.t, n=200)
```

Comment:
Why max_attempts=1?  This sounds like it could cause a race condition on slower machines.